### PR TITLE
[Merged by Bors] - feat: define finitely-semisimple linear maps

### PR DESCRIPTION
**Pull request by ocfnash**
The motivation is to be able to apply `IsFinitelySemisimple.maxGenEigenspace_eq_eigenspace` to symmetric maps in infinite dimensions. Such maps are not in general semisimple but are semisimple on each invariant finite-dimensional subspace.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Hom/Set.lean [hunk 1]
```diff
@@ -159,6 +160,16 @@ instance subsingleton_of_wellFoundedGT' [LinearOrder β] [WellFoundedGT β] [Pre
 
 instance unique_of_wellFoundedGT [LinearOrder α] [WellFoundedGT α] : Unique (α ≃o α) := Unique.mk' _
 
+/-- An order isomorphism between lattices induces an order isomorphism between corresponding
+interval sublattices. -/
+protected def Iic [Lattice α] [Lattice β] (e : α ≃o β) (x : α) :
```
Comment (jcommelin) [inline Mathlib/Order/Hom/Set.lean:165]: Are there similar isoms for other `Ixx`?

File: Mathlib/Order/LatticeIntervals.lean [hunk 2]
```diff
@@ -126,6 +126,16 @@ protected lemma isCompl_iff [Lattice α] [BoundedOrder α] {x y : Iic a} :
     IsCompl x y ↔ Disjoint (x : α) (y : α) ∧ (x : α) ⊔ (y : α) = a := by
   rw [_root_.isCompl_iff, Iic.disjoint_iff, Iic.codisjoint_iff]
 
+protected lemma complementedLattice_iff [Lattice α] [BoundedOrder α] :
```
Comment (jcommelin) [inline Mathlib/Order/LatticeIntervals.lean:129]: Are there similar results for other `Ixx`?

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Order/Hom/Set.lean (hunk 1 above)
Comment (ocfnash) [reply Mathlib/Order/Hom/Set.lean:165]: Yes, I've added `Ici` and `Icc`.

File: Mathlib/Order/LatticeIntervals.lean (hunk 2 above)
Comment (ocfnash) [reply Mathlib/Order/LatticeIntervals.lean:129]: There are, but looking closely, I see the theory of `Iic` is quite a bit more developed than the others already so I think it should be a a separate undertaking to even things out here.

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**